src/components/edit/exchange-update/requests-overview.js

```diff
@@ -0,0 +1,171 @@
+import { useConfig } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import {
+    Button,
+    ButtonStrip,
+    Table,
+    TableHead,
+    TableCellHead,
+    TableRowHead,
+    TableBody,
+    TableRow,
+    DataTableCell,
+} from '@dhis2/ui'
+import PropTypes from 'prop-types'
+import React from 'react'
+import { OU_LEVEL_PREFIX, OU_GROUP_PREFIX } from '../shared/index.js'
+import styles from './requests-overview.module.css'
+
+const getOuText = ({ ouInfo }) => {
+    const orgUnits = ouInfo.filter(
+        ({ id }) =>
+            !id.startsWith(OU_LEVEL_PREFIX) && !id.startsWith(OU_GROUP_PREFIX)
+    )
+    const orgUnitLevels = ouInfo.filter(({ id }) =>
+        id.startsWith(OU_LEVEL_PREFIX)
+    )
+    const orgUnitGroups = ouInfo.filter(({ id }) =>
+        id.startsWith(OU_GROUP_PREFIX)
+    )
+    let orgUnitString = ''
+    if (orgUnits.length === 1) {
+        orgUnitString += orgUnits[0].name
+    }
+    if (orgUnits.length > 1) {
+        orgUnitString += i18n.t('{{firstOuName}} and {{ouLength}} others', {
+            firstOuName: ouInfo[0].name,
+            ouLength: ouInfo.length - 1,
+        })
+    }
+    if (orgUnitLevels.length > 0) {
+        orgUnitString += i18n.t('; levels: {{levelCount}}', {
+            levelCount: orgUnitLevels.length,
+            nsSeparator: '-:-',
+        })
+    }
+    if (orgUnitGroups.length > 0) {
+        orgUnitString += i18n.t('; groups: {{groupsCount}}', {
+            groupsCount: orgUnitGroups.length,
+            nsSeparator: '-:-',
+        })
+    }
+    return orgUnitString
+}
+
+const EmptyTableInfo = () => (
+    <DataTableCell colspan="6">
+        <div className={styles.emptyTableWrapper}>
+            <p className={styles.emptyTableHeader}>{i18n.t('No requests')}</p>
+            <p className={styles.emptyTableText}>
+                {i18n.t(
+                    'Click on buttons below to add a new request. Exchanges must contain at least one request.'
```

Comment:
@cooper-joe I'm curious... do we usually try to avoid referring to interface elements like this?

Reply:
I don't know if we generally have a policy, but we sparingly make use of language like this in some other apps (e.g. in [data entry app](https://github.com/dhis2/aggregate-data-entry-app/blob/master/src/data-workspace/entry-screen.js#L53-L57), or in the [users app](https://github.com/dhis2/user-app/blob/master/src/pages/UserList/UserTable.js#L60) [not explicitly mentioning button, but we say "below" and the button that they're supposed to use is right below))

Reply:
Generally, yes, we try to avoid some types of this as it's a form of UI-layout-dependency that can become out of sync with the actual UI. However, it can be reworded slightly to remove the directionality and therefore to avoid the dependency, for example:
`Click the 'Add request' button to add a request to this exchange`.